auraed/src/runtime/cells/cell_service.rs

```diff
@@ -63,11 +63,7 @@ impl CellService {
         info!("CellService: allocate() cell={:?}", cell);
 
         if self.cells.contains(&cell.name)? {
-            return Err(CellsError::Other(anyhow!(
-                "cell '{}' already exists",
-                cell.name
-            ))
-            .into());
+            return Err(CellError::Exists { cell_name: cell.name }.into());
```

Comment:
it frustrates me that this is here and not an error on insert, but i know we already "allocate" the cell so we need to check it first.  should allocate do this check anyway? otherwise we'll be trying to create a duplicate fs directory for this cgroup i think.

Reply:
I'm not sure I understand. We are in `CellService::allocate` here within the check for `self.cells.contains`, but before we call `Cell::allocate`.

Do you mean `Cell::allocate` should check with the system directly? Probably not a bad idea, but if we get there, our cache has become inconsistent with the system (which is something I agree we need to consider a possibility and try to heal).